src/chunk_constraint.c

```diff
@@ -452,14 +401,17 @@ ChunkConstraints *
 ts_chunk_constraint_scan_by_chunk_id(int32 chunk_id, Size num_constraints_hint, MemoryContext mctx)
 {
 	ChunkConstraints *constraints = ts_chunk_constraints_alloc(num_constraints_hint, mctx);
-	int num_found;
+	ScanKeyData scankey[1];
```

Comment:
While I overall like the simplifications provided by this interface, I feel it is a bit odd and non-intuitive that the scankey's declaration and size is now detached from its initialization. We declare it here, and implicitly it needs some specific size, but it is never referenced or used further in this function.

Reply:
changed to the embedded version